src/components/repositories/remote-form.tsx

```diff
@@ -16,12 +16,48 @@ import {
 
 import { WriteOnlyField, HelperText, FileUpload } from 'src/components';
 
-import { DownloadIcon } from '@patternfly/react-icons';
+import {
+  DownloadIcon,
+  ExclamationTriangleIcon,
+  ExclamationCircleIcon,
+} from '@patternfly/react-icons';
 
 import { RemoteType, WriteOnlyFieldType } from 'src/api';
 import { Constants } from 'src/constants';
 import { isFieldSet, isWriteOnly, ErrorMessagesType } from 'src/utilities';
 
+const validateURLHelper = (
+  outsideError: string | undefined,
+  url: string,
+): {
+  validated: 'default' | 'warning' | 'error';
+  helperTextInvalid?: string;
+  helperText?: string;
+} => {
+  if (outsideError) {
+    return { validated: 'error', helperTextInvalid: outsideError };
+  }
+
+  try {
+    const { protocol } = new URL(url);
+    if (protocol === 'http:') {
+      return {
+        validated: 'warning',
+        helperText: t`Consider using a secure URL (https://).`,
+      };
+    }
+
+    if (protocol === 'https:') {
+      return { validated: 'default' };
+    }
+  } catch (_) {}
```

Comment:
Just this lint error, otherwise LGTM :+1: :)

```suggestion
  } catch (_) {
    // fallthrough
  }
```

fixes that :)